setup/stability-extra/multicluster-vpn/setup.sh

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env bash
+
+# Configures an 'external cluster' for the `multicluster-vpn` scenarios.
+# See `README.md` for details.
+#
+# This script is configured via the following set of environment variables (most of the arguments have a reasonable default value and can be left out):
+#
+# HUB                  -- Istio container images repository
+# TAG                  -- Istio container images version tag
+# APPS_NAMESPACE       -- Should be the same as the corresponding namespace in the primary stability cluster
+# ISTIOCTL             -- Custom path to `istioctl`
+# ISTIO_INSTALLER_PATH -- Custom path to Istio Installer (will be downloaded if not provided)
+# KUBECONFIG1          -- `kubeconfig` of the primary stability cluster
+# KUBECONTEXT1         -- context of the primary stability cluster
+# KUBECONFIG2          -- `kubeconfig` of the external cluster that's being configured
+# KUBECONTEXT2         -- context of the external cluster that's being configured
+#
+# A complete example:
+#
+# ```
+# $ ISTIO_INSTALLER_PATH=~/works/installer \
+#   ISTIOCTL="go run istio.io/istio/istioctl/cmd/istioctl" \
+#   HUB=gcr.io/istio-release \
+#   TAG=master-latest-daily \
+#   KUBECONTEXT1=gke_istio-test-230101_us-central1-a_mc-s \
+#   KUBECONTEXT2=gke_istio-test-230101_us-west1-a_mc-uswest \
+#   ./setup.sh
+# ```
+
+set -e
+
+HUB="${HUB:-gcr.io/istio-release}"
+TAG="${TAG:-master-latest-daily}"
+APPS_NAMESPACE="${APPS_NAMESPACE:-istio-stability-multicluster-vpn}"
+ISTIOCTL="${ISTIOCTL:-istioctl}"
+
+# KUBECONFIG: path to a kubeconfig file
+KUBECONFIG1="${KUBECONFIG1:-${HOME}/.kube/config}"
+KUBECONFIG2="${KUBECONFIG2:-${HOME}/.kube/config}"
```

Comment:
Won't this default to them both being the same cluster? maybe only the first should default?

Reply:
I see you have the check later so I guess it doesn't matter

Reply:
I find my self using either of the two options most often: 1) default kubeconfig, different context names; 2) same context name, two different kubeconfig files (in case I'm generating one on the fly)

So having such defaults allows to pass less arguments on average. And that check below protects the user from the most basic errors such as not passing either of the arguments, etc.

Reply:
Oh never cool, I didn't think about that